app/Models/Port.php

```diff
@@ -61,30 +61,34 @@ public static function boot()
      */
     public function getLabel()
     {
-        $os = $this->device->os;
+        $label = '';
 
-        if (\LibreNMS\Config::getOsSetting($os, 'ifname')) {
-            $label = $this->ifName;
-        } elseif (\LibreNMS\Config::getOsSetting($os, 'ifalias')) {
-            $label = $this->ifAlias;
-        }
+        if (isset($this->device) && is_object($this->device)) {
```

Comment:
This is wrong, it prevents the code that's not dependent on the device to run.
The code I posted on discord does the correct thing.

Reply:
This should be in a own pr together with the changes above

Reply:
Again, we shouldn't have ports that don't have a device.

Reply:
> This is wrong, it prevents the code that's not dependent on the device to run.
> The code I posted on discord does the correct thing. 
> 

Sorry for this, I must saw your code. I'll right on to it...

Reply:
> This should be in a own pr together with the changes above

Ok, let's solidify this code and all fixes, and I'll separate this into two new PRs.